diff --git a/node_modules/@radix-ui/react-select/dist/index.js b/node_modules/@radix-ui/react-select/dist/index.js
index 486fc4e..840617a 100644
--- a/node_modules/@radix-ui/react-select/dist/index.js
+++ b/node_modules/@radix-ui/react-select/dist/index.js
@@ -414,7 +414,9 @@ var SelectContentImpl = React.forwardRef(
           if (pointerMoveDelta.x <= 10 && pointerMoveDelta.y <= 10) {
             event.preventDefault();
           } else {
-            if (!content.contains(event.target)) {
+            // Patch: We use `composedPath` to account for shadow DOM.
+            const target = event.composedPath()[0] ?? event.target;
+            if (!content.contains(target)) {
               onOpenChange(false);
             }
           }
diff --git a/node_modules/@radix-ui/react-select/dist/index.mjs b/node_modules/@radix-ui/react-select/dist/index.mjs
index 0ff1d62..d55bb86 100644
--- a/node_modules/@radix-ui/react-select/dist/index.mjs
+++ b/node_modules/@radix-ui/react-select/dist/index.mjs
@@ -347,7 +347,9 @@ var SelectContentImpl = React.forwardRef(
           if (pointerMoveDelta.x <= 10 && pointerMoveDelta.y <= 10) {
             event.preventDefault();
           } else {
-            if (!content.contains(event.target)) {
+            // Patch: We use `composedPath` to account for shadow DOM.
+            const target = event.composedPath()[0] ?? event.target;
+            if (!content.contains(target)) {
               onOpenChange(false);
             }
           }
